### PR TITLE
Fluid pipes 2

### PR DESCRIPTION
**Pull request by brachy84**
Changes how fluid pipes work.

- Every pipe can store one or multiple fluids
- the total of all fluids in all pipes is stored in the pipe net
- all pipes with a valid `IFluidHandler` neighbour are ticking and will try push fluids
- Fluid filter covers are respected between pipes (but not pumps)
- fluids are drained and filled into and from the Network
- the net then handles the distribution between each pipe
- the fluid pipe network itself is also ticking
- when a pipe gets a fluid from a side, it won't push anything to that side for 20 ticks   (opinions?)
- the dmg the player receives on collision should be the average (not weighted) of all contained fluids

Fluids should be equally splitted among all pipe. The fluids will be equally splitted among all neighbours of a pipe. This means that the handler of a pipe with 2 neighbours will get half the fluid then a handler whitchs pipe only has one neighbour
(handler is a `IFluidHandler` and neighbour is a adjacent `IFluidHandler`)

The end result is a bit weird, but overall it works. I tried to go for a solution that has the functionality of GT5U pipes whithout having each pipe ticking. Another issue with GT5U pipes is sloshing. With their current implementation i couldn't think of a good solution to prevent this. This is why i came up with the current behaviour. It might not be perfect, but at this point I don't fucking care anymore. And I will not rewrite them again. If you have suggestions for improvements, please let me know.

for testing purpose you can right click on pipes to see the capacity and it's content

### CODE REVIEW

**Review by DStrand1 — COMMENTED** [review]
Overall the functionality of these seems much better, but it seems to do transfer half of the current fluid stored in the pipe instead of up to half of the total capacity each tick, which causes some strange transfer rates and would make these pipes a bit worse to use since their transfer rate is dependent on their filled amount

**Comment by DStrand1** [comment]
Also as an aside, I think we need to change the default for "Allow input from output side" back to false, otherwise these can behave strangely (not in a "bad functionality" way, they work as expected given the default, but it could easily be confusing to players)

**Comment by DStrand1** [comment]
Also as far as I can tell, the multi-fluid pipes work great now!

**Comment by DStrand1** [comment]
As another aside, we may want to consider adding long distance fluid pipes now since our normal pipes tick

**Review by DStrand1 — COMMENTED** [review]
After these last commits, behavior seems closer but still pretty similar. In my setup, I used a Huge Bronze Pipe (4800L/s) and an input tank with 4B of water going to another tank. Its expected that it should all be transferred within 1 second, but it still has the "tapering off" effect where its halving the amount of fluid it has to insert each operation

**Comment by brachy84** [comment]
Ah, it's probably because the fluid gets evend out after each insertion tick in the net. Pipes can only push fluids that they contain.
In my testing the 4B was instantly put into the pipes, but the last ~300mb into the other tank where a little slow

**Review by htmlcsjs — APPROVED** [review]
Looks good from the gameplay pov and a quick glance over the code

**Comment by DStrand1** [comment]
Behavior seems good, but from my testing, at least the multifluid pipes are not respecting filters used as covers placed on the pipe itself instead of the output tank

File: src/main/java/gregtech/common/pipelike/fluidpipe/net/FluidPipeNet.java [hunk 1]
```diff
@@ -1,93 +1,210 @@
...
+                int amount = Math.min(stack.amount, stack1.amount);
+                if (!doDrain || amount <= 0)
+                    return amount;
+                stack1.amount -= amount;
+                if (stack1.amount <= 0) {
+                    iterator.remove();
+                    fluidsToRemove.put(stack1, pos);
+                } else if (!silent)
+                    dirtyStacks.put(stack1, pos);
+                GTLog.logger.info("Drained {} * {} from net", stack.getLocalizedName(), amount);
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/common/pipelike/fluidpipe/net/FluidPipeNet.java:66]: Can the info level logs in this class be changed to `debug` level logs?

**Review by TechLord22 — APPROVED** [review]

**Pull request merged**